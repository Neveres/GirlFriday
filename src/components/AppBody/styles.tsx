import { css } from '@emotion/react'

export const appBodyContainer = css`
  display: inline-block;
  margin-left: 210px;
  height: 100vh;
`
