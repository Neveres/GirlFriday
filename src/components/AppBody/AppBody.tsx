/** @jsxImportSource @emotion/react */
import React, { useContext, useMemo } from 'react'
import { AppContext, SideBar } from 'src/components'
import { menuItems } from 'src/Routes'
import { appBodyContainer } from './styles'

export interface IAppBodyProps {
  pathname: string
  children?: React.ReactElement
}

const AppBody: React.FC<IAppBodyProps> = ({ pathname, children }) => {
  const {
    state: { showSideBar },
  } = useContext(AppContext)

  const { withSideBar } = useMemo(
    () =>
      menuItems.find(({ path }) => path === pathname) as GirlFriday.NavBarItem,
    [pathname],
  )

  return (
    <div css={appBodyContainer}>
      {children}
      {showSideBar && withSideBar && <SideBar />}
    </div>
  )
}

export default AppBody
