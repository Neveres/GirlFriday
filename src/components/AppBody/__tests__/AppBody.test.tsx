import React from 'react'
import { create } from 'react-test-renderer'

import AppBody from '../AppBody'

describe('AppBody', () => {
  test('should render well with required props', () => {
    const props = {
      pathname: '',
      children: <>test</>,
    }
    const testRenderer = create(<AppBody {...props} />)
    expect(testRenderer.toJSON()).toMatchSnapshot()
  })
})
