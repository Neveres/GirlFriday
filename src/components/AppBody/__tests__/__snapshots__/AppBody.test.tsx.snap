// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBody should render well with required props 1`] = `
<div
  className="css-j9dwpg-AppBody"
>
  test
</div>
`;
